examples/11_hens/hens_perturbation.py

```diff
@@ -0,0 +1,163 @@
+
+
+class HENSPerturbation:
+    """Initialize the HemisphericCentredBredVector perturbation method for obtaining a calibrated ensemble as described in Huge Ensembles Part I: Design of Ensemble Weather Forecasts using Spherical Fourier Neural Operators https://arxiv.org/abs/2408.03100.
+
+    Parameters
+    ----------
+    model : PrognosticModel
+        The prognostic model to be used for ensemble forecasting
+    data : DataSource
+        Data source for obtaining initial conditions
+    start_time : datetime
+        The initial time for the ensemble forecast
+    skill_path : str
+        Path to the file containing forecast skill scores for noise amplification
+    noise_amplification : float
+        Base amplification factor for the noise
+    perturbed_var : str | list[str]
+        Variable(s) to be perturbed in the ensemble
+    integration_steps : int
+        Number of integration steps for the bred vector calculation
+    """
+
+    def __init__(
+        self,
+        model: PrognosticModel,
+        data: DataSource,
+        start_time: datetime,
+        skill_path: str,
+        noise_amplification: float,
+        perturbed_var: str | list[str],
+        integration_steps: int,
+    ) -> None:
+
+        noise_amp_seed = self.get_noise_vector(
+            model,
+            skill_path=skill_path,
+            noise_amplification=noise_amplification,
+            vars=perturbed_var,
+        )
+        noise_amp_iter = self.get_noise_vector(
+            model,
+            skill_path=skill_path,
+            noise_amplification=noise_amplification,
+        )
+
+        seed_perturbation = CorrelatedSphericalGaussian(
+            noise_amplitude=noise_amp_seed,
+            sigma=1.0,
+            length_scale=5.0e5,
+            time_scale=48.0,
+        )
+
+        self.perturbation = HemisphericCentredBredVector(
+            model=model,
+            data=data,
+            # time=start_time,
+            noise_amplitude=noise_amp_iter,
+            integration_steps=integration_steps,  # use cfg.breeding_steps
+            seeding_perturbation_method=seed_perturbation,
+        )
+
+        return
+
+    def get_noise_vector(
+        self,
+        model: PrognosticModel,
+        skill_path: str = None,
+        noise_amplification: float = 1.0,
+        vars: str | list[str] | None = None,
+        lead_time: int = 48,
+    ) -> Tensor:
+        """Generate a noise vector for the HemisphericCentredBredVector perturbation method.
+
+        This function creates a variable-specific noise vector based on forecast skill scores.
+        The noise amplitude for each variable is scaled according to its forecast skill at the
+        specified lead time, with variables not in the perturbation list set to zero.
+
+        Parameters
+        ----------
+        model : PrognosticModel
+            The prognostic model used for forecasting
+        skill_path : str, optional
```

Comment:
Consider defining a package that points to the hugging face repo... so it will auto download and cache for the user

Reply:
Will track here: https://github.com/NVIDIA/earth2studio/issues/286